### PR TITLE
Skeleton shopping cart block

### LINKED ISSUE #1295: Cart Block: Block skeleton registered (php/js)
For this issue:

* [ ] Cart block registered that is just the basic skeleton for the block registered php side and js side. Just use placeholder for content right now - components will come in future iterations.
* [ ] Should have "Full Cart" and "Empty Cart" modes (again with just placeholder content).
* [ ] ~The block should only be available via the cart page.~ This has been split out to #1305

### PR DESCRIPTION
**Pull request by haszari**
_In progress_

Fixes #1295

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

### How to test the changes in this Pull Request:

1.
2.
3.

### Changelog

> Add suggested changelog entry here.

### CODE REVIEW

File: assets/js/blocks/cart-checkout/cart/block.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { Placeholder } from '@wordpress/components';
+
+/**
+ * Component to handle edit mode of "Featured Product".
```
Comment (senadir) [inline assets/js/blocks/cart-checkout/cart/block.js]: simple typo here
```suggestion
 * Component to handle edit mode of "Cart Block".
```

File: assets/js/blocks/cart-checkout/cart/example.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { DEFAULT_HEIGHT } from '@woocommerce/block-settings';
+import { previewProducts } from '@woocommerce/resource-previews';
+
+export const example = {
```
Comment (senadir) [inline assets/js/blocks/cart-checkout/cart/example.js]: This block won’t have this many attributes, I think we can skip a separate `example.js` file and just keep the `example` in the `index.js` file

File: assets/js/blocks/cart-checkout/cart/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { InnerBlocks } from '@wordpress/editor';
+import { registerBlockType } from '@wordpress/blocks';
+
+/**
+ * Internal dependencies
+ */
+import Block from './block';
+
+/**
+ * Register and run the "Featured Product" block.
```
Comment (senadir) [inline assets/js/blocks/cart-checkout/cart/index.js]:
```suggestion
 * Register and run the "Cart" block.
```

File: assets/js/blocks/cart-checkout/cart/index.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		align: [ 'wide', 'full' ],
+		html: false,
+	},
+	attributes: {},
+
+	/**
+	 * Renders and manages the block.
+	 */
+	edit( props ) {
+		return <Block { ...props } />;
```
Comment (senadir) [inline assets/js/blocks/cart-checkout/cart/index.js]: We will end up moving this to a separate `Edit` component later that will call `<Block>` inside it with a `<Disabled>` like other blocks
https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/master/assets/js/blocks/active-filters/edit.js#L84

File: bin/webpack-helpers.js [hunk 5]
```diff
@@ -123,6 +125,9 @@ const mainEntry = {
 	'active-filters': './assets/js/blocks/active-filters/index.js',
 	'block-error-boundary':
 		'./assets/js/base/components/block-error-boundary/style.scss',
+
+	// cart & checkout blocks
```
Comment (senadir) [inline bin/webpack-helpers.js]: Cart & Checkout blocks should be treated as separate blocks, both in folder and naming, I would suggest we have a top-level block folder for each one of them.

File: src/BlockTypes/Cart.php [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+				'editor_script'   => 'wc-' . $this->block_name,
+				'editor_style'    => 'wc-block-editor',
+				'style'           => 'wc-block-style',
+				'script'          => 'wc-' . $this->block_name . '-frontend',
+			)
+		);
+	}
+
+	/**
+	 * Append frontend scripts when rendering the Product Categories List block.
```
Comment (senadir) [inline src/BlockTypes/Cart.php]:
```suggestion
	 * Append frontend scripts when rendering the Cart block.
```

**Comment by nerrad** [comment]
I addressed all the feedback in the latest pull.  Also,

- frontend script and registration is added.  I realize we're initially focusing on the edit context but it's easy enough to add and puts the skeleton in place for when we start looking at doing the frontend.

File: bin/webpack-helpers.js (hunk 5 above)
Comment (haszari) [reply bin/webpack-helpers.js]: That was my thinking - our blocks folder is getting hard to navigate, and naming is hard. Folders for the main groups can help break it up :)

**Pull request merged**